### PR TITLE
feat(ui): Persist variables control bar preferences

### LINKED ISSUE #12845: Default Variables Control Bar to "open" in dashboards
When a user opens a dashboard that has variables, the Variables Control Bar should be automatically opened

### LINKED ISSUE #12846: Save Variable Control Bar open/close state for dashboards
As a user
If I open/close the Variable Control Bar on a dashboard with variables
Then I would like it to be in the same open/close state on all dashboards with variables when I leave and return to the app

### PR DESCRIPTION
**Pull request by ischolten**
Closes #12845
Closes #12846

_Briefly describe your proposed changes:_
Save user preference for open or closed variables control bar in local storage.
default to open.

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] http/swagger.yml updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/dashboards/components/DashboardPage.tsx [hunk 1]
```diff
@@ -168,14 +166,13 @@ class DashboardPage extends Component<Props, State> {
...
             handleChooseAutoRefresh={handleChooseAutoRefresh}
             handleChooseTimeRange={this.handleChooseTimeRange}
             handleClickPresentationButton={handleClickPresentationButton}
-            toggleVariablesControlBar={this.toggleVariablesControlBar}
-            isShowingVariablesControlBar={isShowingVariablesControlBar}
+            toggleVariablesControlBar={onToggleShowVariablesControls}
+            isShowingVariablesControlBar={showVariablesControls}
           />
-          {isShowingVariablesControlBar && (
+          {showVariablesControls && !!dashboard && (
```
Comment (OfTheDelmer) [inline ui/src/dashboards/components/DashboardPage.tsx]: nice

File: ui/src/mockState.tsx [hunk 2]
```diff
@@ -25,7 +26,8 @@ const localState = {
       duration: '15m',
     },
   ],
-  variables: initialState(),
+  variables: initialVariablesState(),
```
Comment (OfTheDelmer) [inline ui/src/mockState.tsx]: would we need to worry about sensitive data in variables?

File: ui/src/shared/components/RefreshingView.tsx [hunk 3]
```diff
@@ -68,39 +71,45 @@ class RefreshingView extends PureComponent<Props, State> {
...
     const {submitToken} = this.state
 
     return (
-      <TimeSeries
-        inView={inView}
-        submitToken={submitToken}
-        queries={this.queries}
-        key={manualRefresh}
-        variables={this.variableAssignments}
+      <SpinnerContainer
```
Comment (OfTheDelmer) [inline ui/src/shared/components/RefreshingView.tsx]: Nice

File: ui/src/mockState.tsx (hunk 2 above)
Comment (ischolten) [reply ui/src/mockState.tsx]: the initial state of variables doesnt have much in it. plus this is just mocks

**Pull request merged**